### ISSUE TITLE
Steam Crashes in Linux and causes kernel panic.

### ISSUE THREAD

**Issue by som-cs** [ValveSoftware/steam-for-linux#11303]
#### Your system information
- System Info: https://gist.github.com/som-cs/27e1a69fc428a21a72f38a6075ba9f16

* Steam client version (build number or date): 
**Issue started on beta yesterday then i uninstalled and reinstalled steam and the issue persisted.**

    ```
    Steam Beta Branch:  Stable Client
    Steam Version:  1726604483
    Steam Client Build Date:  Tue, Sep 17 9:03 PM UTC -00:00
    Steam Web Build Date:  Tue, Sep 17 12:12 AM UTC -00:00
    Steam API Version:  SteamClient021
    ```

* Distribution (e.g. Ubuntu): 6.6.47-1-MANJARO
*
* Opted into Steam client beta?: [Yes/No]: No

* Have you checked for system updates?: [Yes/No]: Yes

* Steam Logs: [generate by running this command in a terminal `tar -zcvf ~/Desktop/steam-logs.tar.gz ~/.steam/steam/logs`]: 
[steam-logs.tar.gz](https://github.com/user-attachments/files/17168559/steam-logs.tar.gz)

* GPU: Nvidia 3060 Laptop

#### Please describe your issue in as much detail as possible:
- I was in a game of dota 2 suddenly the game hanged. When i looked around I found out that steam is also not responding. 
- I tried multiple times to open steam but it crashed and always after crashing none of the applications could be used and system had to be restarted. Even the running application crashed like firefox and system monitor.
- After a long time i was able to get steam running but in the process i had removed the dota 2 game. I started downloading the game and suddenly the steam client crashed and the same crash issues started. 70% of the times it kept ending in kernel panic. Sometime I was able to reboot the system.

#### Steps for reproducing this issue:

1. Open Steam
2. Start Downloading dota 2
3. crash

**At first I was doing `prime-run steam` but later on when i am using steam normally the issue still persists.**

journalctl -f output: https://gist.github.com/som-cs/16bcaaf2d4ae868f8170b779adaea506

latest crash logs: https://gist.github.com/som-cs/dec5c5f7dac46a50692cd4ce16078fcc

Steam ticket: 	HT-7V2Q-Y5NH-XNKD

**Comment by som-cs** [comment]
journalctl: https://gist.github.com/som-cs/ef2ba7d9688484d5ed35f723116f1561

**Comment by smcv** [comment]
A kernel panic is always a bug in the kernel or a kernel-space driver: if the kernel is working as intended, Steam shouldn't be able to cause a panic, even if it was actively trying to. It's possible that the Steam crash might be a side-effect, rather than the root cause.

From the other information you provided, it seems that unrelated processes like `appimagelauncherd.service` are also crashing, which would suggest that something is wrong outside Steam.

These crash dumps in the Journal:

```
Sep 29 18:51:58 caramell kernel:  ? __die+0x23/0x70
Sep 29 18:51:58 caramell kernel:  ? page_fault_oops+0x174/0x530
Sep 29 18:51:58 caramell kernel:  ? exc_page_fault+0x7f/0x180
Sep 29 18:51:58 caramell kernel:  ? asm_exc_page_fault+0x26/0x30
Sep 29 18:51:58 caramell kernel:  ? filemap_get_entry+0x8e/0x140
Sep 29 18:51:58 caramell kernel:  ? filemap_get_entry+0x72/0x140
Sep 29 18:51:58 caramell kernel:  __filemap_get_folio+0x39/0x2f0
Sep 29 18:51:58 caramell kernel:  ext4_da_write_begin+0x129/0x260 [ext4 75091d576b0947e7de591be92e65d7aa9ca7da30]
Sep 29 18:51:58 caramell kernel:  generic_perform_write+0xd9/0x240
Sep 29 18:51:58 caramell kernel:  ext4_buffered_write_iter+0x67/0x100 [ext4 75091d576b0947e7de591be92e65d7aa9ca7da30]
Sep 29 18:51:58 caramell kernel:  do_iter_readv_writev+0xf6/0x160
```

```
`0x23/0x70
Sep 29 18:52:03 caramell kernel:  ? page_fault_oops+0x174/0x530
Sep 29 18:52:03 caramell kernel:  ? mempool_alloc+0x87/0x1b0
Sep 29 18:52:03 caramell kernel:  ? exc_page_fault+0x7f/0x180
Sep 29 18:52:03 caramell kernel:  ? asm_exc_page_fault+0x26/0x30
Sep 29 18:52:03 caramell kernel:  ? mpage_process_page_bufs+0x47/0x160 [ext4 75091d576b0947e7de591be92e65d7aa9ca7da30]
Sep 29 18:52:03 caramell kernel:  mpage_map_and_submit_buffers+0x1b5/0x250 [ext4 75091d576b0947e7de591be92e65d7aa9ca7da30]
Sep 29 18:52:03 caramell kernel:  ext4_do_writepages+0x804/0xca0 [ext4 75091d576b0947e7de591be92e65d7aa9ca7da30]
Sep 29 18:52:03 caramell kernel:  ext4_writepages+0xaf/0x180 [ext4 75091d576b0947e7de591be92e65d7aa9ca7da30]
Sep 29 18:52:03 caramell kernel:  do_writepages+0x72/0x1e0
```

are in filesystem code, so I would suggest using a known-good OS image (perhaps a "live" recovery image) to make sure that there is no filesystem corruption.

**Comment by som-cs** [comment]
Yesterday i was trying to fix the issue and uninstalled steam and nvidia drivers to see if the issue was actually there. It toom a long time, way more then it does with steam but the kernel panic did occur. This started some doubts in me. Then during shutdown I saw 0.3% blocks were not having any neighbours. I ran the fdck checks and also badblocks but nothing came up. Everything came back clean.

As a last resort I just started reinstalling Manjaro and then at 28% the installation just hanged for hours. TWICE. So yes this is a hardware issue and i have ordered some new hard drives for me. This was in no way a steam or nvidia issue.


Thank you for your help.
